Emby.Server.Implementations/Dto/DtoService.cs

```diff
@@ -1102,12 +1102,14 @@ private void AttachBasicFields(BaseItemDto dto, BaseItem item, BaseItem owner, D
 
             if (options.ContainsField(ItemFields.LocalTrailerCount))
             {
-                allExtras ??= item.GetExtras().ToArray();
-                dto.LocalTrailerCount = allExtras.Count(i => i.ExtraType == ExtraType.Trailer);
-
                 if (item is IHasTrailers hasTrailers)
                 {
-                    dto.LocalTrailerCount += hasTrailers.GetTrailerCount();
+                    dto.LocalTrailerCount = hasTrailers.GetTrailerCount();
+                }
+                else
+                {
+                    allExtras ??= item.GetExtras().ToArray();
+                    dto.LocalTrailerCount = allExtras.Count(i => i.ExtraType == ExtraType.Trailer);
```

Comment:
```suggestion
                    dto.LocalTrailerCount = (allExtras ?? item.GetExtras()).Count(i => i.ExtraType == ExtraType.Trailer);
```

Reply:
If we are sure we won't use `allExtras` (as a cache?) below in the future. Otherwise we have to touch this line again.

Reply:
But this is now :)

Reply:
Isn't it like:
```cs
if (!allExtras) allExtras = item.GetExtras().ToArray();
```
We could remove `.ToArray()` though (from both lines).

Reply:
?? is the null coalescing operator. It will return the left hand side if it's not null, otherwise it returns the right hand side.

We can't remove the toarray since it will enumerate twice if both ifs are entered.